src/mame/machine/hp2640_tape.h

```diff
@@ -0,0 +1,96 @@
+// license:BSD-3-Clause
+// copyright-holders: F. Ulivi
+/*********************************************************************
+
+    hp2640_tape.h
+
+    Tape subsystem of HP264x terminals
+
+*********************************************************************/
+
+#ifndef MAME_MACHINE_HP2640_TAPE_H
+#define MAME_MACHINE_HP2640_TAPE_H
+
+#pragma once
+
+#include "machine/hp_dc100_tape.h"
+
+class hp2640_tape_device : public device_t
+{
+public:
+	// construction/destruction
+	hp2640_tape_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	// callbacks
+	auto irq() { return m_irq_handler.bind(); }
+	auto led_0() { return m_led0_handler.bind(); }
+	auto led_1() { return m_led1_handler.bind(); }
+
+	// I/O
+	void    command_w(uint8_t cmd);
+	uint8_t status_r ();
+	void    data_w   (uint8_t data);
+	uint8_t data_r   ();
+	uint8_t poll_r   () const;
+
+protected:
+	// device-level overrides
+	virtual void device_add_mconfig(machine_config &config) override;
+	virtual void device_start() override;
+	virtual void device_reset() override;
+	virtual void device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr) override;
+
+private:
+	devcb_write_line m_irq_handler;
+	devcb_write_line m_led0_handler;
+	devcb_write_line m_led1_handler;
+
+	// 0 is left hand drive
+	// 1 is right hand drive
+	required_device_array<hp_dc100_tape_device , 2> m_drives;
+
+	// State
+	unsigned m_selected_drive;  // U35-10
```

Comment:
You should use an explicitly sized type for this so it can be registered for save states portably.